rmw_connext_shared_cpp/include/rmw_connext_shared_cpp/shared_functions.hpp

```diff
@@ -229,6 +229,13 @@ wait(const char * implementation_identifier,
           }
         }
         if (fixed) {
+          // Reset the fixed guard conditions to avoid being woken up
+          // immediately next time.
+          DDSGuardCondition * guard = static_cast<DDSGuardCondition *>((*attached_conditions)[i]);
+          retcode = guard->set_trigger_value(DDS_BOOLEAN_FALSE);
+          if (retcode != DDS_RETCODE_OK) {
+            RMW_SET_ERROR_MSG("failed to set trigger value");
```

Comment:
I see, then maybe it should go out to stderr directly? Or maybe we can just ignore it, since our error handling will print out to stderr if the message is overwritten without being read first, e.g. (problem I'm currently debugging):

```
[rmw|error_handling.c:39] error string being overwritten: wait set is invalid, at /Users/william/ros2_ws/src/ros2/rcl/rcl/src/rcl/wait.c:382
[rmw|error_handling.c:39] error string being overwritten: failed to create topic, at /Users/william/ros2_ws/src/ros2/rmw_opensplice/rmw_opensplice_cpp/src/rmw_publisher.cpp:137
/Users/william/ros2_ws/src/ros2/rcl/rcl/test/rcl/test_subscription.cpp:169: Failure
Value of: ret
  Actual: 1
Expected: 0
failed to create topic, at /Users/william/ros2_ws/src/ros2/rmw_opensplice/rmw_opensplice_cpp/src/rmw_publisher.cpp:137, at /Users/william/ros2_ws/src/ros2/rcl/rcl/src/rcl/publisher.c:81
```

You can see two messages were ignored for some reason.